File: OJ/Luogu/P1290.cpp
#include<iostream>
#include<cstdio>
#include<cmath>

using namespace std;

bool findDivisor(int a, int b) {
  int _min = min(a, b);
  int i = 2;
  for (; i <= _min; i ++) {
    if (a % i == 0 && b % i == 0) {
      return true;
    }
  }
  if (i > _min) {
    return false;
  }
}


int main() {
  int n;
  cin >> n;
  while(n--) {
    int a, b;
    cin >> a >> b;
    if (findDivisor(a, b)) {
      cout << "Ollie wins" << endl;
    } else {
      cout << "Stan wins" << endl;
    }
  }
  return 0;
}

File: OJ/Luogu/P1290.js
function findDivisor(a, b) {
    let min = Math.min(a, b)
    let i = 2;
    for (; i <= min; i ++) {
        if (a % i === 0 && b % i === 0) {
            console.log(i);
            return true
        }
    }
    if (i > min) return false;
}

function P1290() {
    if (!findDivisor(24, 15)) {
        console.log("Ollie wins");
    } else {
        console.log("Stan wins");
    }
}